relay/cache/cached_accessor.go

```diff
@@ -86,14 +89,13 @@ func NewCachedAccessor[K comparable, V any](
 
 func newAccessResult[V any]() *accessResult[V] {
 	result := &accessResult[V]{
-		wg: sync.WaitGroup{},
+		sem: semaphore.NewWeighted(1),
 	}
-	result.wg.Add(1)
+	result.sem.TryAcquire(1)
```

Comment:
This always succeeds, should it just Acquire()?

Reply:
This will always succeed either way, since this happens right after the semaphore is created and since no other goroutine has access to the semaphore yet. The reason why I use `TryAcquire()` is that it doesn't return an error. If I use `Acquire()` then I have to add error handling, even though it's impossible for an error to be returned.

With this in mind, are you ok with `TryAcquire(1)` here?

Reply:
Does `_ = result.sem.Acquire()` work (with comment that this should always succeed)?

TryAquire() returns a boolean, so it's the same in the sense that the return value is ignored.

So both will work, the difference I think is that TryAcquire reads like it doesn't care about whether it has grabbed the resource.

Reply:
I've switched this to `_ = result.sem.Acquire(context.Background(), 1)`